### PR TITLE
Remove BPF mount if Cilium is specified as Network Plugin

### PR DESCRIPTION
**Pull request by mattstam**
When copied over from AKS-engine, this was initially here to to mount the BPFFS for Cilium. 

Since then, Cilium has been updated to use systemd mount for the BPFFS. This gets done as part of the cilium-node-init startup script.
```                 
                 ...
                  mount | grep "/sys/fs/bpf type bpf" || {
                    echo "Mounting BPF filesystem..."
                    mount bpffs /sys/fs/bpf -t bpf
                    echo "Installing BPF filesystem mount"
                    cat >/tmp/sys-fs-bpf.mount <<EOF
                 ...
```

### CODE REVIEW

**Comment by xuto2** [comment]
can you update unit test?

**Comment by matt-stam** [comment]
> can you update unit test?

Hi Tony. I don't think is any unit tests for Cilium currently. And we're actually moving away from setting up non kubenet/azurecni/calico, so it may make more sense to just slowly phase these out from AgentBaker as they seemed to just have been copied from aks-engine.

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 1]
```diff
@@ -230,15 +230,6 @@ write_files:
     }
 {{end}}
 
-{{if HasCiliumNetworkPlugin }}
-- path: /etc/systemd/system/sys-fs-bpf.mount
```
Comment (paulgmiller) [inline parts/linux/cloud-init/nodecustomdata.yml:234]: this file is still used somewhere else or should we remove it too?

Comment (mattstam) [reply parts/linux/cloud-init/nodecustomdata.yml:234]: Removed it now, seems to have only been used if `HasCiliumNetworkPlugin` was true

**Comment by paulgmiller** [comment]
Think we need to regnerate some files after these changes. Charlie should be quite familiar with this now.

**Comment by charliedmcb** [comment]
> Think we need to regnerate some files after these changes. Charlie should be quite familiar with this now.

Yup, @mattstam you need to run ```make generate``` and commit the diff.

**Comment by charliedmcb** [comment]
> > Think we need to regnerate some files after these changes. Charlie should be quite familiar with this now.
> 
> Yup, @mattstam you need to run `make generate` and commit the diff.
> 
> From a previous conversation I had with Erik Wu:
> "every time you made changes in that parts/ folder (i.e shell scripts, yaml files, etc), you should run make generate"

Also, from Erik:
"That make target also does a shell-check at the end. That might fail if you're not on a mac or ubuntu i believe in which case you can ignore. The upstream github repo will re-run shell-check anyways." 

BTW, the check was failing for me on WSL, and I don't believe its been fixed yet.

**Review by xuto2 — APPROVED** [review]
Thank you for cleaning up!

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 2]
```diff
@@ -268,11 +268,6 @@ configureCNI() {
     retrycmd_if_failure 120 5 25 modprobe br_netfilter || exit $ERR_MODPROBE_FAIL
     echo -n "br_netfilter" > /etc/modules-load.d/br_netfilter.conf
     configureCNIIPTables
-    {{if HasCiliumNetworkPlugin}}
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_config.sh:271]: there is also a check of HasCiliumNetworkPlugin in L393 when we enabling kubelet, shall we remove it as well?

https://github.com/Azure/AgentBaker/pull/575/files#diff-4f9504893023c819ba00993d7db2a80f9883ef64f249e6e61b06d18094f3d425R388

Comment (xuto2) [reply parts/linux/cloud-init/artifacts/cse_config.sh:271]: also can we remove HasCiliumNetworkPolicy and HasCiliumNetworkPlugin from baker.go, L463- L466?

Comment (mattstam) [reply parts/linux/cloud-init/artifacts/cse_config.sh:271]: removed them

**Comment by paulgmiller** [comment]
Looks like you picked up some conflicts. Lets try and fix them on monday and I'll help merge this in.

**Comment by mattstam** [comment]
This PR confuses Github so retrying in https://github.com/Azure/AgentBaker/pull/638

**Pull request closed without merging**